### PR TITLE
Fix serialization tests run failed

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: I25cb4b691ed6401ec749ab3b5152b1796ccca244

### CODE REVIEW

File: hugegraph-cassandra/src/main/java/com/baidu/hugegraph/backend/store/cassandra/CassandraTables.java [hunk 1]
```diff
@@ -397,7 +401,7 @@ protected void deleteEdgesByLabel(CassandraSessionPool.Session session,
             final String OTHER_VERTEX = formatKey(HugeKeys.OTHER_VERTEX);
 
             // Query edges by label index
-            Select select = QueryBuilder.select().from(this.table());
+            Select select = QueryBuilder.select().from(this.labelQueryTable());
```
Comment (javeme) [inline hugegraph-cassandra/src/main/java/com/baidu/hugegraph/backend/store/cassandra/CassandraTables.java]: prefer labelIndexTable()
also update `count > COMMIT_DELETE_BATCH - 2` to `count >= COMMIT_DELETE_BATCH - 2`

File: hugegraph-scylladb/src/main/java/com/baidu/hugegraph/backend/store/scylladb/ScyllaDBTablesWithMV.java [hunk 2]
```diff
@@ -36,24 +35,183 @@
...
-            }
+            Object value = cq.condition(key);
+            return value != null && cq.allSysprop() &&
+                   conditions.size() == 1 &&
+                   cq.containsCondition(key, Condition.RelationType.EQ);
         }
         return false;
     }
 
+    protected static boolean isQueryByLabel(Query query) {
```
Comment (javeme) [inline hugegraph-scylladb/src/main/java/com/baidu/hugegraph/backend/store/scylladb/ScyllaDBTablesWithMV.java]: set private

File: hugegraph-scylladb/src/main/java/com/baidu/hugegraph/backend/store/scylladb/ScyllaDBTablesWithMV.java [hunk 3]
```diff
@@ -36,24 +35,183 @@
...
+        String cql = String.format("DROP MATERIALIZED VIEW IF EXISTS %s",
+                                   mvName);
+        session.execute(cql);
+    }
+
+    public static class PropertyKey extends CassandraTables.PropertyKey {
+
+        private final String MV_NAME2PK = mvNameTable(this.table());
+
+        public PropertyKey() {
```
Comment (javeme) [inline hugegraph-scylladb/src/main/java/com/baidu/hugegraph/backend/store/scylladb/ScyllaDBTablesWithMV.java]: remove unused constructor

File: hugegraph-scylladb/src/main/java/com/baidu/hugegraph/backend/store/scylladb/ScyllaDBTablesWithMV.java [hunk 4]
```diff
@@ -36,24 +35,183 @@
...
+            }
+            return super.query2Select(table, query);
+        }
+    }
+
+    public static class VertexLabel extends CassandraTables.VertexLabel {
+
+        private final String MV_NAME2VL = mvNameTable(this.table());
+
+        public VertexLabel() {
```
Comment (javeme) [inline hugegraph-scylladb/src/main/java/com/baidu/hugegraph/backend/store/scylladb/ScyllaDBTablesWithMV.java]: ditto

File: hugegraph-test/src/main/resources/fast-methods.filter [hunk 5]
```diff
@@ -61,6 +61,11 @@ org.apache.tinkerpop.gremlin.structure.GraphTest.shouldRemoveVertices: Random UU
 org.apache.tinkerpop.gremlin.structure.GraphTest.shouldRemoveEdges: Random UUID as edge label
 org.apache.tinkerpop.gremlin.structure.TransactionTest.shouldExecuteWithCompetingThreads: Hang the tests because property 'test' have both String and long value
 
+## vertex properties doesn't have nested structures
+org.apache.tinkerpop.gremlin.structure.SerializationTest.GraphSONTest.shouldSerializeVertex: Vertex properties doesn't have nested structures
+org.apache.tinkerpop.gremlin.structure.SerializationTest.GraphSONTest.shouldSerializePath: Vertex properties doesn't have nested structures
+org.apache.tinkerpop.gremlin.structure.SerializationTest.GraphSONTest.shouldSerializeTree: Vertex properties doesn't have nested structures
+
```
Comment (javeme) [inline hugegraph-test/src/main/resources/fast-methods.filter]: also add to methods.filter

File: hugegraph-core/src/main/java/com/baidu/hugegraph/io/HugeGraphSONModule.java [hunk 6]
```diff
@@ -89,9 +97,9 @@
         TYPE_DEFINITIONS.put(EdgeLabel.class, "EdgeLabel");
         TYPE_DEFINITIONS.put(IndexLabel.class, "IndexLabel");
 
-        // HugeGraph vertex/edge serializer
+        // HugeGraph vertex serializer
         TYPE_DEFINITIONS.put(HugeVertex.class, "HugeVertex");
-        TYPE_DEFINITIONS.put(HugeEdge.class, "HugeEdge");
+//        TYPE_DEFINITIONS.put(HugeEdge.class, "HugeEdge");
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/io/HugeGraphSONModule.java]: add some comments why not register HugeEdgeSerializer

File: hugegraph-cassandra/src/main/java/com/baidu/hugegraph/backend/store/cassandra/CassandraStoreProvider.java [hunk 7]
```diff
@@ -52,7 +52,8 @@ public String version() {
          * [1.0] HugeGraph-1328: supports backend table version checking
          * [1.1] HugeGraph-1322: add support for full-text search
          * [1.2] #296: support range sortKey feature
+         * [1.3] Modify the table structure of indexlabel
```
Comment (javeme) [inline hugegraph-cassandra/src/main/java/com/baidu/hugegraph/backend/store/cassandra/CassandraStoreProvider.java]: use a separate commit to fix this bug

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java [hunk 8]
```diff
@@ -71,6 +72,48 @@ public ConditionQuery query(Condition condition) {
...
+    public void validate() {
+        // Check conditions of label
+        List<List<Object>> labelsList = new ArrayList<>();
+        for (Condition condition : this.conditions) {
+            if (!condition.isRelation()) {
+                continue;
+            }
+            Relation relation = (Relation) condition;
+            if (relation.key().equals(HugeKeys.LABEL)) {
+                if (relation.value() instanceof List) {
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java]: should relation be EQ or IN? how about other relations like NEQ?

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java [hunk 9]
```diff
@@ -71,6 +72,48 @@ public ConditionQuery query(Condition condition) {
...
+                    @SuppressWarnings("unchecked")
+                    List<Object> labels = (List<Object>) relation.value();
+                    labelsList.add(labels);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java]: prefer define a labelsSet and call labelsSet.addAll(labels)

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java [hunk 10]
```diff
@@ -71,6 +72,48 @@ public ConditionQuery query(Condition condition) {
...
+                } else {
+                    List<Object> labels = new ArrayList<>();
+                    labels.add(relation.value());
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java]: labelsSet.add(relation.value())

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java [hunk 11]
```diff
@@ -71,6 +72,48 @@ public ConditionQuery query(Condition condition) {
...
+            }
+            if (results.isEmpty()) {
+                break;
+            }
+        }
+        if (results != null && !results.isEmpty()) {
+            this.conditions.add(Condition.in(HugeKeys.LABEL, results));
+        }
+    }
+
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java]: also add a separate commit

File: hugegraph-api/src/main/java/com/baidu/hugegraph/serializer/JsonSerializer.java [hunk 12]
```diff
@@ -51,23 +50,16 @@
...
 public class JsonSerializer implements Serializer {
 
-    private final GraphSONWriter writer;
-
     private static final int BUF_SIZE = 128;
     private static final int LBUF_SIZE = 1024;
 
-    public JsonSerializer(GraphSONWriter writer) {
-        this.writer = writer;
+    private static JsonSerializer INSTANCE = new JsonSerializer();
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/serializer/JsonSerializer.java]: BUF_SIZE and LBUF_SIZE are unused any more?

File: hugegraph-test/src/main/resources/fast-methods.filter [hunk 13]
```diff
@@ -61,6 +61,11 @@ org.apache.tinkerpop.gremlin.structure.GraphTest.shouldRemoveVertices: Random UU
 org.apache.tinkerpop.gremlin.structure.GraphTest.shouldRemoveEdges: Random UUID as edge label
 org.apache.tinkerpop.gremlin.structure.TransactionTest.shouldExecuteWithCompetingThreads: Hang the tests because property 'test' have both String and long value
 
+## vertex properties doesn't have nested structures
```
Comment (javeme) [inline hugegraph-test/src/main/resources/fast-methods.filter]: vertex properties doesn't have nested structures with HugeVertexSerializer

File: hugegraph-api/src/main/java/com/baidu/hugegraph/serializer/JsonSerializer.java (hunk 12 above)
Comment (Linary) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/serializer/JsonSerializer.java]: LBUF_SIZE still in using

**Pull request merged**